bridge/docs/bridging.md

```diff
@@ -34,3 +34,91 @@ Again, a withdrawfee of 1 TFT will be taken, so make sure you send a larger amou
+
+#### Overview of the TFChain Minting events
+1. `tftBridgeModule.MintTransactionProposed`: A bridge validator has proposed a mint transaction upon witness a stellar deposit.
+2. `tftBridgeModule.MintTransactionVoted`: Other bridge validators witness same stellar deposit and voted for the mint proposal.
+3. `tftBridgeModule.MintCompleted`: Enough bridge validators' votes was collected and the tokens minted to the target address successfully.
+
+#### When a Refund-on-Stellar occurs?
+
+A refund on Stellar occurs when one of the following conditions is met:
+
+- The deposited amount is lower than the deposit fee.
+- The memo message is empty.
+- The transaction contains more than one payment.
+- The memo is not formatted correctly.
+- The grid type is not supported (not one of grid, farm, node, or entity) or not found.
+
+### STELLAR -> TFCHAIN (Refund-on-Stellar flow)
+In this section, we look into the details of what happens when the a Stellar deposit can not be processed sue to a validation problem.
+
+1. A transaction is received on the bridge Stellar account and witnessed by a bridge validator.
+
+2. The transaction information undergoes validation. here we assume it failed because of one of the violations mentioned in the previous section, so a refund flow is initiated by the bridge by calling TFTBridgeModule `create_refund_transaction_or_add_sig` extrinsic to propose a `RefundTransaction` and store the details in `RefundTransactions` storage map alongside with bridge validator signature. `RefundTransactionsignatureAdded` and `RefundTransactionCreated` events.
+
+3. Other bridge validators also provides their signature for that transaction proposal by calling same extrinsic.
+
+4. If the majority (more than the half) of bridge validators provided their signature for a refund transaction, a `RefundTransactionReady` event is emitted as well. This check happens every time the `create_refund_transaction_or_add_sig` extrinsic is executed by validator call.
+
+5. The bridge will handle the event and query TFChain storage for the `RefundTransaction` details and the validators’ signatures. It will create a multi-signatures Stellar transaction with a [MEMO](https://developers.stellar.org/docs/encyclopedia/memos) of `RETURN` type containing the hash of the refunded transaction and submit it to Stellar network. If submitted successfully, it will call `set_refund_transaction_executed` extrinsic (which removes the `RefundTransaction` from the `RefundTransactions` storage and adds it to `ExecutedRefundTransactions`) then emit `RefundTransactionProcessed` event.
+
+
+#### Overview of the TFChain Refund events
+1. `tftBridgeModule.RefundTransactionCreated`: A bridge validator has proposed a Refund-on-Stellar transaction upon witness a stellar deposit with invalid or missing cross-chain transfer information.
```

Comment:
A bridge validator proposed a Refund-on-Stellar transaction after being the first to witness a stellar deposit with invalid or missing cross-chain transfer information.